### PR TITLE
Simplify test workspace construction

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@roslyn-ide plz review.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide Plz review

File: src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.cs [hunk 1]
```diff
@@ -939,8 +939,8 @@ private static bool ContainsVisualBasicKeywords(string input)
             private static async Task<TestWorkspace> CreateWorkspaceFromFileAsync(string file, bool isVisualBasic, ParseOptions parseOptions, CompilationOptions compilationOptions)
             {
                 return isVisualBasic ?
-                    await VisualBasicWorkspaceFactory.CreateWorkspaceFromFileAsync(file, (VB.VisualBasicParseOptions)parseOptions, (VB.VisualBasicCompilationOptions)compilationOptions) :
-                    await CSharpWorkspaceFactory.CreateWorkspaceFromFileAsync(file, (CS.CSharpParseOptions)parseOptions, (CS.CSharpCompilationOptions)compilationOptions);
+                    await TestWorkspace.CreateVisualBasicAsync(file, (VB.VisualBasicParseOptions)parseOptions, (VB.VisualBasicCompilationOptions)compilationOptions) :
+                    await TestWorkspace.CreateCSharpAsync(file, (CS.CSharpParseOptions)parseOptions, (CS.CSharpCompilationOptions)compilationOptions);
```
Comment (DustinCampbell) [inline src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.cs]: Are the casts to language-specific options necessary anymore?

File: src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb [hunk 2]
```diff
@@ -35,7 +35,7 @@ Namespace Microsoft.CodeAnalysis.Editor.Implementation.Diagnostics.UnitTests
                            </Project>
                        </Workspace>
 
-            Using workspace = TestWorkspaceFactory.CreateWorkspace(test)
+            Using workspace = TestWorkspace.CreateWorkspace(test)
```
Comment (DustinCampbell) [inline src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb]: Shouldn't `CreateWorkspace(...)` be renamed to `Create(...)` to be consistent with `CreateAsync(...)`?

File: src/VisualStudio/Core/Test/Diagnostics/ExternalDiagnosticUpdateSourceTests.vb [hunk 3]
```diff
@@ -27,7 +27,7 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Diagnostics
 
         <Fact>
         Public Async Function TestExternalDiagnostics_RaiseEvents() As Task
-            Using workspace = Await CSharpWorkspaceFactory.CreateWorkspaceFromFileAsync(String.Empty)
+            Using workspace = Await TestWorkspace.CreateCSharpAsync(String.Empty)
```
Comment (DustinCampbell) [inline src/VisualStudio/Core/Test/Diagnostics/ExternalDiagnosticUpdateSourceTests.vb]: Does `String.Empty` really need to be passed everytime? Seems like that parameter should be optional, or there should be another factory method (e.g. `TestWorkspace.CreateEmptyCSharpAsync(...)`).

**Comment by DustinCampbell** [comment]
Thanks for doing this work. It's something I'm wanted to clean up for a long time.

**Comment by DustinCampbell** [comment]
:+1: assuming you consider my comments.

File: src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.cs]: Nope.  I'll fix up!

File: src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb]: Yup!  I will address in a followup checkin.  (i'm not done cleaning this all up).

File: src/VisualStudio/Core/Test/Diagnostics/ExternalDiagnosticUpdateSourceTests.vb (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Test/Diagnostics/ExternalDiagnosticUpdateSourceTests.vb]: enh.... i think ti's fine to pass "" or String.empty for this case.

**Pull request merged**